### PR TITLE
feat: [CDS-72361]:hiding-dynamic-provisioner-asg

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Removed dynamic provisioner from Asg

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by litvinovstas** [comment]
retrigger jest

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx [hunk 1]
```diff
@@ -624,7 +625,10 @@ export default function DeployEnvironment({
             gitOpsEnabled={gitOpsEnabled}
           />
         )}
-        {!(isFixed && isEmpty(selectedEnvironments)) && !isMultiEnvironment && CD_NG_DYNAMIC_PROVISIONING_ENV_V2 ? (
+        {!(isFixed && isEmpty(selectedEnvironments)) &&
+        !isMultiEnvironment &&
+        CD_NG_DYNAMIC_PROVISIONING_ENV_V2 &&
+        deploymentType !== ServiceDeploymentType.Asg ? (
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:631]: Let's move this to a function/util.

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:631]: Can move it later as well once the deployment type list starts increasing in size

**Review by vikrant-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good to me @litvinovstas

**Comment by ishantmehtaharness** [comment]
SorryCypress is failing

**Comment by litvinovstas** [comment]
retrigger cypress-rest

**Pull request merged**